scanner/indexer/indexer_test.go

```diff
@@ -0,0 +1,62 @@
+package indexer
+
+import (
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseImageURL(t *testing.T) {
+	tests := []struct {
+		name    string
+		arg     string
+		want    name.Reference
+		wantErr string
+	}{
+		{
+			name:    "empty URL",
+			arg:     "",
+			wantErr: "invalid URL",
+		},
+		{
+			name:    "no schema",
```

Comment:
can you also add one with a scheme but not tag nor digest?

Reply:
Give me specific examples of all the tests you are looking for; otherwise, we might go back and forth here.